Gateway/Response/TransactionAuthorizeHandler.php

```diff
@@ -42,8 +42,8 @@ public function handle(array $handlingSubject, array $response)
         $paymentDO = SubjectReader::readPayment($handlingSubject);
         /** @var Payment $orderPayment */
         $orderPayment = $paymentDO->getPayment();
-        $orderPayment->setAdditionalInformation(self::CHARGE_ID, $response[self::TRANSACTION_ID]);
-        $orderPayment->setTransactionId($response[self::TRANSACTION_ID]);
+        $orderPayment->setAdditionalInformation(self::TRANSACTION_ID, $response[self::TRANSACTION_OBJECT_ID]);
```

Comment:
is there a reason why we are setting this as transaction ID vs charge ID.  if we keep it as charges ID would we be able to remove all the if statements?

Reply:
The main reason was that it helps to keep consistency between the API object name and the attribute name, in case there are further changes to the API down the road (such as converging of the checkout token and transaction object id). Also this would help more easily identify which API is being used without having additional context when troubleshooting.